src/Raven.Server/Documents/Sharding/Comparers/DocumentsComparer.cs

```diff
@@ -10,31 +11,70 @@
 
 namespace Raven.Server.Documents.Sharding.Comparers;
 
-public class DocumentsComparer : IComparer<BlittableJsonReaderObject>
+public sealed class DocumentsComparer : IComparer<BlittableJsonReaderObject>
 {
     private readonly OrderByField[] _orderByFields;
     private readonly bool _extractFromData;
+    private readonly Random[] _randoms;
+    private readonly int[] _orderToRandom;
 
     public DocumentsComparer(OrderByField[] orderByFields, bool extractFromData)
     {
         _orderByFields = orderByFields;
         _extractFromData = extractFromData;
+        int randomCount = RandomComparersCount(_orderByFields);
+        (_randoms, _orderToRandom) = randomCount == 0 
+            ? (null, null) 
+            : GetRandom(randomCount, _orderByFields);
     }
 
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    private static int RandomComparersCount(OrderByField[] orderByField)
+    {
+        int randomComparerCount = 0;
+        for (int cmpId = 0; cmpId < orderByField.Length; ++cmpId)
+        {
+            if (orderByField[cmpId].OrderingType == OrderByFieldType.Random)
+                randomComparerCount++;
+        }
+        
+        return randomComparerCount;
+    }
+
+    private (Random[], int[]) GetRandom(int randomCount, OrderByField[] orderByFields)
+    {
+        var randoms = new Random[randomCount];
```

Comment:
simpler to just create `orderByFields.Length` length here and just put null?

Reply:
Yes, done